# Challenge Telecom X
Telecom X - Análisis de Evasión de Clientes
Has sido contratado como asistente de análisis de datos en Telecom X y formarás parte del proyecto "Churn de Clientes". La empresa enfrenta una alta tasa de cancelaciones y necesita comprender los factores que llevan a la pérdida de clientes.

Tu desafío será recopilar, procesar y analizar los datos, utilizando Python y sus principales bibliotecas para extraer información valiosa. A partir de tu análisis, el equipo de Data Science podrá avanzar en modelos predictivos y desarrollar estrategias para reducir la evasión.

¿Qué vas a practicar?
✅ Importar y manipular datos desde una API de manera eficiente.<br>
✅ Aplicar los conceptos de ETL (Extracción, Transformación y Carga) en la preparación de los datos.<br>
✅ Crear visualizaciones estratégicas para identificar patrones y tendencias.<br>
✅ Realizar un Análisis Exploratorio de Datos (EDA) y generar un informe con insights relevantes.<br>
<br>
¡Ahora es tu turno! 🚀 Usa tus conocimientos para transformar datos en información estratégica y ayudar a Telecom X a retener más clientes.

## Extracción
Empezamos por la extracción de datos directamente desde el enlace

In [106]:
# Comenzamos importando todas las librerías necesarias.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [107]:
# Nos conectamos a la fuente de datos directa:
url = "https://raw.githubusercontent.com/ingridcristh/challenge2-data-science-LATAM/refs/heads/main/TelecomX_Data.json"
datos = pd.read_json(url)
# Vemos las primeras filas del DataFrame.
datos.head()

,customerID,Churn,customer,phone,internet,account
0,0002-ORFBO,No,"{'gender': 'Female', 'SeniorCitizen': 0, 'Part...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'One year', 'PaperlessBilling': '..."
1,0003-MKNFE,No,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'Yes'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
2,0004-TLHLJ,Yes,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
3,0011-IGKFF,Yes,"{'gender': 'Male', 'SeniorCitizen': 1, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
4,0013-EXCHZ,Yes,"{'gender': 'Female', 'SeniorCitizen': 1, 'Part...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."


Nos pudimmos conectar sin problema a los datos originales y cargar un DataFrame.<br>
### Conoce los datos
Podemos ver inmediatamente que las columnas "customer", "phone", "internet" y "account" están anidadas en formato .json. Usaremos .info() para saber más de los datos.

In [108]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   customerID  7267 non-null   object
 1   Churn       7267 non-null   object
 2   customer    7267 non-null   object
 3   phone       7267 non-null   object
 4   internet    7267 non-null   object
 5   account     7267 non-null   object
dtypes: object(6)
memory usage: 340.8+ KB


Tenemos 7267 registros y 6 columnas de las cuales ninguna tiene valores vacíos y todas son Dtype object. Podríamos empezar a realizar cambios en éste punto, pero lo dejaremos para la siguiente sección. Lo que sí podemos hacer por ahora es verificar que la columna "customerID" no tenga valores repetidos, de esta forma nos aseguramos que cada entrada corresponda a un cliente diferente.

In [109]:
# Verificación de unicidad de customerID. Si el resultado es menor que 7267, hay IDs duplicados.
datos["customerID"].nunique()

7267

Ahora podemos crear sub DataFrames para conocer los valores que están anidados.

In [120]:
# Creamos un df por cada columna que es un JSON anidado.
customer = pd.json_normalize(datos["customer"])
phone = pd.json_normalize(datos["phone"])
internet = pd.json_normalize(datos["internet"])
account = pd.json_normalize(datos["account"])

Inspeccionamos la información de cada DF

In [121]:
customer.head(), customer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   gender         7267 non-null   object
 1   SeniorCitizen  7267 non-null   int64 
 2   Partner        7267 non-null   object
 3   Dependents     7267 non-null   object
 4   tenure         7267 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 284.0+ KB


(   gender  SeniorCitizen Partner Dependents  tenure
 0  Female              0     Yes        Yes       9
 1    Male              0      No         No       9
 2    Male              0      No         No       4
 3    Male              1     Yes         No      13
 4  Female              1     Yes         No       3,
 None)

El DF "customer" cuenta con 5 columnas, de las cuales ninguna tiene valores nulos. la columna "SeniorCitizen" es un 0 o 1 como valores booleanos, hay que verificar si efectivamente son los únicos valores que tiene. Se asume que la columna "tenure" es la cantidad de años que el cliente ha tenido empleo fijo.

In [122]:
# Corroboramos que las columnas solo tengan los valores esperados.
print(customer["gender"].unique())
print(customer["SeniorCitizen"].unique())
print(customer["Partner"].unique())
print(customer["Dependents"].unique())
print(customer["tenure"].unique())

['Female' 'Male']
[0 1]
['Yes' 'No']
['Yes' 'No']
[ 9  4 13  3 71 63  7 65 54 72  5 56 34  1 45 50 23 55 26 69 11 37 49 66
 67 20 43 59 12 27  2 25 29 14 35 64 39 40  6 30 70 57 58 16 32 33 10 21
 61 15 44 22 24 19 47 62 46 52  8 60 48 28 41 53 68 51 31 36 17 18 38 42
  0]


Los valores son los esperados, no hay que hacer modificaciones.

In [123]:
phone.head(), phone.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 2 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   PhoneService   7267 non-null   object
 1   MultipleLines  7267 non-null   object
dtypes: object(2)
memory usage: 113.7+ KB


(  PhoneService MultipleLines
 0          Yes            No
 1          Yes           Yes
 2          Yes            No
 3          Yes            No
 4          Yes            No,
 None)

In [124]:
print(phone["PhoneService"].unique())
print(phone["MultipleLines"].unique())


['Yes' 'No']
['No' 'Yes' 'No phone service']


El DF "phone cuenta con 2 columnas, ambas tienen todos los valores. No hay modificaciones por hacer.

In [125]:
internet.head(), internet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   InternetService   7267 non-null   object
 1   OnlineSecurity    7267 non-null   object
 2   OnlineBackup      7267 non-null   object
 3   DeviceProtection  7267 non-null   object
 4   TechSupport       7267 non-null   object
 5   StreamingTV       7267 non-null   object
 6   StreamingMovies   7267 non-null   object
dtypes: object(7)
memory usage: 397.5+ KB


(  InternetService OnlineSecurity OnlineBackup DeviceProtection TechSupport  \
 0             DSL             No          Yes               No         Yes   
 1             DSL             No           No               No          No   
 2     Fiber optic             No           No              Yes          No   
 3     Fiber optic             No          Yes              Yes          No   
 4     Fiber optic             No           No               No         Yes   
 
   StreamingTV StreamingMovies  
 0         Yes              No  
 1          No             Yes  
 2          No              No  
 3         Yes             Yes  
 4         Yes              No  ,
 None)

In [126]:
print(internet["InternetService"].unique())
print(internet["OnlineSecurity"].unique())
print(internet["OnlineBackup"].unique())
print(internet["DeviceProtection"].unique())
print(internet["TechSupport"].unique())
print(internet["StreamingTV"].unique())
print(internet["StreamingMovies"].unique())

['DSL' 'Fiber optic' 'No']
['No' 'Yes' 'No internet service']
['Yes' 'No' 'No internet service']
['No' 'Yes' 'No internet service']
['Yes' 'No' 'No internet service']
['Yes' 'No' 'No internet service']
['No' 'Yes' 'No internet service']


El DF "internet" cuenta con 7 columnas de las cuales ninguna tiene valores nulos. No hay modificaciones por hacer.

In [127]:
account.head(), account.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Contract          7267 non-null   object 
 1   PaperlessBilling  7267 non-null   object 
 2   PaymentMethod     7267 non-null   object 
 3   Charges.Monthly   7267 non-null   float64
 4   Charges.Total     7267 non-null   object 
dtypes: float64(1), object(4)
memory usage: 284.0+ KB


(         Contract PaperlessBilling     PaymentMethod  Charges.Monthly  \
 0        One year              Yes      Mailed check             65.6   
 1  Month-to-month               No      Mailed check             59.9   
 2  Month-to-month              Yes  Electronic check             73.9   
 3  Month-to-month              Yes  Electronic check             98.0   
 4  Month-to-month              Yes      Mailed check             83.9   
 
   Charges.Total  
 0         593.3  
 1         542.4  
 2        280.85  
 3       1237.85  
 4         267.4  ,
 None)

In [128]:
print(account["Contract"].unique())
print(account["PaperlessBilling"].unique())
print(account["PaymentMethod"].unique())        

['One year' 'Month-to-month' 'Two year']
['Yes' 'No']
['Mailed check' 'Electronic check' 'Credit card (automatic)'
 'Bank transfer (automatic)']


El DF "account" cuenta con 5 columnas de las cuales "Charges.Total" es un object y hay que transformarlo en float64.

Con ésto terminamos la extracción de datos, estamos listos para empezar a transformarlos según nuestras necesidades.

## Transformación
Ya que todas nuestras bases están cargadas, podemos empezar a transformar los datos según nos convenga, para ello, primero voy a concatenar la información.

In [55]:
# De los datos originales, solo queremos conservar la columna "Churn"
# Queremos concatenar las tablas horizontalmente, éso es el eje 1.
df = pd.concat([datos["Churn"], customer, phone, internet, account], axis=1)
df.head()

,Churn,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,Charges.Monthly,Charges.Total
0,No,Female,0,Yes,Yes,9,Yes,No,DSL,No,Yes,No,Yes,Yes,No,One year,Yes,Mailed check,65.6,593.3
1,No,Male,0,No,No,9,Yes,Yes,DSL,No,No,No,No,No,Yes,Month-to-month,No,Mailed check,59.9,542.4
2,Yes,Male,0,No,No,4,Yes,No,Fiber optic,No,No,Yes,No,No,No,Month-to-month,Yes,Electronic check,73.9,280.85
3,Yes,Male,1,Yes,No,13,Yes,No,Fiber optic,No,Yes,Yes,No,Yes,Yes,Month-to-month,Yes,Electronic check,98.0,1237.85
4,Yes,Female,1,Yes,No,3,Yes,No,Fiber optic,No,No,No,Yes,Yes,No,Month-to-month,Yes,Mailed check,83.9,267.4


In [56]:
# Verificamos la información del nuevo DataFrame.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Churn             7267 non-null   object 
 1   gender            7267 non-null   object 
 2   SeniorCitizen     7267 non-null   int64  
 3   Partner           7267 non-null   object 
 4   Dependents        7267 non-null   object 
 5   tenure            7267 non-null   int64  
 6   PhoneService      7267 non-null   object 
 7   MultipleLines     7267 non-null   object 
 8   InternetService   7267 non-null   object 
 9   OnlineSecurity    7267 non-null   object 
 10  OnlineBackup      7267 non-null   object 
 11  DeviceProtection  7267 non-null   object 
 12  TechSupport       7267 non-null   object 
 13  StreamingTV       7267 non-null   object 
 14  StreamingMovies   7267 non-null   object 
 15  Contract          7267 non-null   object 
 16  PaperlessBilling  7267 non-null   object 


Aquí corroboramos que ninguna columna tiene datos nulos. Ahora verficaremos los datos numéricos, si todos están enlistados como tal y si son del tipo correcto.<br>
* SeniorCitizen: int64, es correcto.
* tenure: int64, es correcto.
* Charges.Monthly: float64, es correcto.
* Charges.Total: object, es falso. Hay que cambiar el tipo de dato a float64, ya que es un valor numérico con decimales.

Toca verificar que los datos de cada columna son lo esperado.

In [57]:
# Verificamos si en la columna "Churn", todos los valores son "Yes" o "No".
df["Churn"].unique()

array(['No', 'Yes', ''], dtype=object)

Notamos que hay 3 tipos de datos, "No", "Yes" y  ''. Debemos sustituir '' por np.NaN y quitarlo de la base ya que no nos sirve pues no sabesmos si son cancelaciones o no.

In [62]:
# Convertimos nuestros valores vacíos a NaN para poder trabajar con ellos.
df["Churn"] = df["Churn"].replace('', np.nan)
df["Churn"].unique()

array(['No', 'Yes'], dtype=object)

In [63]:
# Retiramos las filas con valores NaN en la columna "Churn".
df.dropna(subset=["Churn"], inplace=True)
df["Churn"].unique()

array(['No', 'Yes'], dtype=object)

In [64]:
# Verificamos cuántas entradas nos quedan.
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7043 entries, 0 to 7266
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Churn             7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 
 17  

Nos quedamos con 7043 entradas que sabemos si cancelaron o no su servicio

In [65]:
# Intentamos transformar el dato a float64.
df["Charges.Total"].astype("float64")

ValueError: could not convert string to float: ' '

El error nos indica que hay entradas con valores que no se pueden convertir en numero, en este caso fue un espacio vacío.

In [66]:
df["Charges.Total"].replace(' ', np.nan).astype("float64")

0        593.30
1        542.40
2        280.85
3       1237.85
4        267.40
         ...   
7262     742.90
7263    1873.70
7264      92.75
7265    4627.65
7266    3707.60
Name: Charges.Total, Length: 7043, dtype: float64

In [67]:
# La celda pasada funcionó, por lo que podemos guardar la variable y volvemos a verificar el DataFrame.
df["Charges.Total"] = df["Charges.Total"].replace(' ', np.nan).astype("float64")
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7043 entries, 0 to 7266
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Churn             7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 
 17  

In [68]:
df.head()

,Churn,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,Charges.Monthly,Charges.Total
0,No,Female,0,Yes,Yes,9,Yes,No,DSL,No,Yes,No,Yes,Yes,No,One year,Yes,Mailed check,65.6,593.30
1,No,Male,0,No,No,9,Yes,Yes,DSL,No,No,No,No,No,Yes,Month-to-month,No,Mailed check,59.9,542.40
2,Yes,Male,0,No,No,4,Yes,No,Fiber optic,No,No,Yes,No,No,No,Month-to-month,Yes,Electronic check,73.9,280.85
3,Yes,Male,1,Yes,No,13,Yes,No,Fiber optic,No,Yes,Yes,No,Yes,Yes,Month-to-month,Yes,Electronic check,98.0,1237.85
4,Yes,Female,1,Yes,No,3,Yes,No,Fiber optic,No,No,No,Yes,Yes,No,Month-to-month,Yes,Mailed check,83.9,267.40


Nuestro DataFrame está listo para empezar el análisis.

In [69]:
df.describe()

,SeniorCitizen,tenure,Charges.Monthly,Charges.Total
count,7043.000000,7043.000000,7043.000000,7032.000000
mean,0.162147,32.371149,64.761692,2283.300441
std,0.368612,24.559481,30.090047,2266.771362
min,0.000000,0.000000,18.250000,18.800000
25%,0.000000,9.000000,35.500000,401.450000
50%,0.000000,29.000000,70.350000,1397.475000
75%,0.000000,55.000000,89.850000,3794.737500
max,1.000000,72.000000,118.750000,8684.800000


A partir de este punto, podemos manipular los datos para conseguir la información que queremos.<br>
Como lo que nos interesa es saber por qué la gente está cancelando, crearemos un DataFrame con solo los datos de las personas que han cancelado.

In [91]:
# Filtramos los valores de las personas que sí cancelaron el servicio.
churn = df[df["Churn"]=="Yes"]
churn.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1869 entries, 2 to 7263
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Churn             1869 non-null   object 
 1   gender            1869 non-null   object 
 2   SeniorCitizen     1869 non-null   int64  
 3   Partner           1869 non-null   object 
 4   Dependents        1869 non-null   object 
 5   tenure            1869 non-null   int64  
 6   PhoneService      1869 non-null   object 
 7   MultipleLines     1869 non-null   object 
 8   InternetService   1869 non-null   object 
 9   OnlineSecurity    1869 non-null   object 
 10  OnlineBackup      1869 non-null   object 
 11  DeviceProtection  1869 non-null   object 
 12  TechSupport       1869 non-null   object 
 13  StreamingTV       1869 non-null   object 
 14  StreamingMovies   1869 non-null   object 
 15  Contract          1869 non-null   object 
 16  PaperlessBilling  1869 non-null   object 
 17  

Observamos que tenemos 1869 personas que han cancelado de los 7043 datos totales. Ésto representa un $26.53\%$ de la base.<br>
Como ya filtramos las personas que sí cancelaron, podemos eliminar la columna de "Churn".


In [92]:
churn =churn.drop(columns=["Churn"])
churn.head()

,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,Charges.Monthly,Charges.Total
2,Male,0,No,No,4,Yes,No,Fiber optic,No,No,Yes,No,No,No,Month-to-month,Yes,Electronic check,73.9,280.85
3,Male,1,Yes,No,13,Yes,No,Fiber optic,No,Yes,Yes,No,Yes,Yes,Month-to-month,Yes,Electronic check,98.0,1237.85
4,Female,1,Yes,No,3,Yes,No,Fiber optic,No,No,No,Yes,Yes,No,Month-to-month,Yes,Mailed check,83.9,267.40
18,Male,1,No,No,45,Yes,No,DSL,Yes,No,Yes,No,No,Yes,One year,No,Credit card (automatic),62.7,2791.50
19,Male,1,No,No,1,No,No phone service,DSL,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,25.1,25.10


Notamos que el índice mantiene la numeración previa, como éste es un nuevo DF, le pondremos sus propios índices.

In [93]:
churn = churn.reset_index()
churn = churn.drop(columns=["index"])
churn.head()

,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,Charges.Monthly,Charges.Total
0,Male,0,No,No,4,Yes,No,Fiber optic,No,No,Yes,No,No,No,Month-to-month,Yes,Electronic check,73.9,280.85
1,Male,1,Yes,No,13,Yes,No,Fiber optic,No,Yes,Yes,No,Yes,Yes,Month-to-month,Yes,Electronic check,98.0,1237.85
2,Female,1,Yes,No,3,Yes,No,Fiber optic,No,No,No,Yes,Yes,No,Month-to-month,Yes,Mailed check,83.9,267.40
3,Male,1,No,No,45,Yes,No,DSL,Yes,No,Yes,No,No,Yes,One year,No,Credit card (automatic),62.7,2791.50
4,Male,1,No,No,1,No,No phone service,DSL,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,25.1,25.10


## Carga y Análisis
Podemos hacer un análisis por cada columna, sin embargo ya que la empresa da servicios, nos enfocaremos en los servicios ofrecidos.

In [105]:
churn["gender"].sum()

'MaleMaleFemaleMaleMaleFemaleFemaleFemaleMaleMaleFemaleMaleMaleMaleMaleMaleFemaleMaleFemaleFemaleFemaleFemaleMaleFemaleMaleFemaleMaleFemaleFemaleMaleFemaleMaleMaleMaleMaleFemaleMaleFemaleFemaleMaleFemaleMaleFemaleMaleMaleMaleFemaleMaleFemaleFemaleMaleMaleMaleMaleFemaleMaleFemaleFemaleFemaleMaleFemaleFemaleFemaleFemaleMaleMaleFemaleMaleFemaleFemaleMaleFemaleFemaleMaleFemaleMaleFemaleMaleFemaleMaleFemaleFemaleFemaleFemaleMaleMaleMaleFemaleFemaleMaleFemaleFemaleFemaleMaleFemaleFemaleFemaleMaleFemaleMaleFemaleMaleMaleMaleMaleFemaleFemaleFemaleMaleMaleMaleFemaleMaleFemaleMaleMaleMaleFemaleMaleFemaleMaleMaleFemaleFemaleFemaleMaleMaleFemaleFemaleMaleMaleFemaleMaleMaleFemaleFemaleMaleMaleFemaleFemaleFemaleFemaleFemaleMaleMaleMaleFemaleMaleMaleMaleFemaleMaleMaleMaleMaleFemaleMaleMaleFemaleMaleMaleMaleMaleFemaleFemaleFemaleMaleFemaleFemaleFemaleFemaleMaleFemaleFemaleMaleMaleFemaleMaleFemaleFemaleMaleFemaleMaleMaleFemaleMaleMaleMaleFemaleMaleMaleMaleMaleFemaleMaleFemaleMaleMaleFemaleFemaleFemaleM

## Informe Final